lib/dataset/groupby.cpp

```diff
@@ -422,6 +420,22 @@ GroupBy<Dataset> groupby(const Dataset &dataset, const Variable &key,
   throw except::DimensionError("Size of Group-by key is incorrect.");
 }
 
+Variable extract(const Variable &var, const Variable &condition) {
```

Comment:
`extract` is maybe not a good name. We already use it for the C++ equivalent of `dict.pop`. Maybe `select`?

Reply:
I chose this because of https://numpy.org/doc/stable/reference/generated/numpy.extract.html, but you may be right.